packages/components/src/dropdown-menu/index.js

```diff
@@ -43,6 +43,13 @@ function DropdownMenu( {
 		}
 	}
 
+	let activeControl = false;
+	flatMap( controlSets, ( controlSet ) => {
```

Comment:
It seems like a very specific implementation of the heading dropdown. Can you think of an alternative approach where `DropdownMenu` stays unmodified so we could leave this component as is? Ideally, all those CSS styles are applied directly to `HeadingToolbar` components or alternatively, we modify SVG icons to take the number as param.

Reply:
Maybe, but that would complicate this PR. I tried to follow the same pattern in the interest of pushing the feature forward. We do already use `isActive` in the `DropdownMenu` so I don't feel I've introduced specific Heading related behavior, just some logic to determine the active `DropdownMenu` element.

I will look into implementing the heading icons differently and get rid of the subscript in the heading component but we'll see. I wonder if there is any benefit of still using subscript for these kinds of icons?